compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingPatterns.cpp

```diff
@@ -175,45 +152,56 @@ struct MaterializeInterfaceBindingEncoding
   matchAndRewrite(IREE::HAL::InterfaceBindingSubspanOp subspanOp,
                   OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    auto resultType = llvm::dyn_cast<IREE::TensorExt::DispatchTensorType>(
+    auto origResultType = llvm::dyn_cast<IREE::TensorExt::DispatchTensorType>(
         subspanOp.getResult().getType());
-    if (!resultType) {
+    if (!origResultType) {
       return rewriter.notifyMatchFailure(
           subspanOp,
           "expected result type to be !iree_tensor_ext.dispatch.tensor");
     }
-    auto boundTensorType =
-        llvm::dyn_cast<RankedTensorType>(resultType.getBoundType());
-    if (!boundTensorType) {
+    auto origBoundTensorType =
+        llvm::dyn_cast<RankedTensorType>(origResultType.getBoundType());
+    if (!origBoundTensorType) {
       return rewriter.notifyMatchFailure(
           subspanOp, "bound type is not a RankedTensorType");
     }
 
-    auto convertedBoundType = getTypeConverter()->convertType(boundTensorType);
-    if (convertedBoundType == boundTensorType) {
-      return rewriter.notifyMatchFailure(subspanOp, "bound type already valid");
-    }
-
     auto *typeConverter = static_cast<const MaterializeEncodingTypeConverter *>(
         getTypeConverter());
+    auto convertedResultType =
+        typeConverter->convertType<IREE::TensorExt::DispatchTensorType>(
+            origResultType);
+    if (origResultType == convertedResultType) {
```

Comment:
this can also fail and return nullptr, no?

Reply:
in the most general case, type conversion can fail. I think your type converter is guaranteed not to?